server/src/main/java/org/apache/gravitino/server/web/rest/MetadataObjectPolicyOperations.java

```diff
@@ -0,0 +1,375 @@
+import org.apache.gravitino.dto.responses.NameListResponse;
+import org.apache.gravitino.dto.responses.PolicyListResponse;
+import org.apache.gravitino.dto.responses.PolicyResponse;
+import org.apache.gravitino.dto.util.DTOConverters;
+import org.apache.gravitino.exceptions.NoSuchPolicyException;
+import org.apache.gravitino.metrics.MetricNames;
+import org.apache.gravitino.policy.Policy;
+import org.apache.gravitino.policy.PolicyDispatcher;
+import org.apache.gravitino.server.web.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("/metalakes/{metalake}/objects/{type}/{fullName}/policies")
+public class MetadataObjectPolicyOperations {
+  private static final Logger LOG = LoggerFactory.getLogger(MetadataObjectPolicyOperations.class);
+
+  private final PolicyDispatcher policyDispatcher;
+
+  @Context private HttpServletRequest httpRequest;
+
+  @Inject
+  public MetadataObjectPolicyOperations(PolicyDispatcher policyDispatcher) {
+    this.policyDispatcher = policyDispatcher;
+  }
+
+  @GET
+  @Path("{policy}")
+  @Produces("application/vnd.gravitino.v1+json")
+  @Timed(name = "get-object-policy." + MetricNames.HTTP_PROCESS_DURATION, absolute = true)
+  @ResponseMetered(name = "get-object-policy", absolute = true)
+  public Response getPolicyForObject(
+      @PathParam("metalake") String metalake,
+      @PathParam("type") String type,
+      @PathParam("fullName") String fullName,
+      @PathParam("policy") String policyName) {
+    LOG.info(
+        "Received get policy {} request for object type: {}, full name: {} under metalake: {}",
+        policyName,
+        type,
+        fullName,
+        metalake);
+
+    try {
+      return Utils.doAs(
+          httpRequest,
+          () -> {
+            MetadataObject object =
+                MetadataObjects.parse(
+                    fullName, MetadataObject.Type.valueOf(type.toUpperCase(Locale.ROOT)));
+
+            // 1. Check if the policy is directly attached to the object.
+            Optional<Policy> policy = getPolicyForObject(metalake, object, policyName);
+            if (policy.isPresent()) {
+              LOG.info(
+                  "Get policy: {} for object type: {}, full name: {} under metalake: {}",
+                  policyName,
+                  type,
+                  fullName,
+                  metalake);
+              return Utils.ok(
+                  new PolicyResponse(DTOConverters.toDTO(policy.get(), Optional.of(false))));
+            }
+
+            // 2. If not found, try to find an inheritable policy from its parents.
+            // First, ensure the target policy exists and is inheritable for this object type.
+            Policy targetPolicy = policyDispatcher.getPolicy(metalake, policyName);
+            if (!targetPolicy.inheritable()
+                || !targetPolicy.supportedObjectTypes().contains(object.type())) {
+              return getPolicyNotFoundResponse(metalake, policyName, type, fullName);
+            }
```

Comment:
Where is the logic to get policy from its parent? You just randomly get a policy by its name. I'm super confused about your logic here. You need to explain more.

Reply:
In fact, getting the policy from its parent node starts from the following `while (parentObject != null)`, and the logic here is `//First, make sure the target policy exists and can be inherited by this object type.`

I will remove this inaccurate comment:`//2. If not found, try to find an inheritable policy from its parents.`